_posts/2024-03-28-fuse-storage-driver.adoc

```diff
@@ -0,0 +1,84 @@
+
+image::/assets/img/fuse-storage-driver/layer-diagram.png[]
+Fig. 1. Example comparison of image layer sizes between vfs and overlay storage drivers. The total size of the image layers created with overlay is about half the size of the image layers created with vfs.
+
+## How can we use fuse-overlayfs in an Eclipse Che cloud development environment?
+
+To use fuse-overlayfs in an Eclipse Che cloud development environment (CDE), the CDE's pod requires access to the `/dev/fuse` device from the host operating system. For Kubernetes, support for this can vary depending on the platform. However, for Openshift, version 4.15 allows access to the `/dev/fuse` device https://docs.openshift.com/container-platform/4.15/release_notes/ocp-4-15-release-notes.html#ocp-4-15-nodes-dev-fuse[without any cluster modifications]. Here's the documentation on how to https://eclipse.dev/che/docs/stable/end-user-guide/accessing-fuse[enable the overlay storage driver for CDEs running on OpenShift].
+
+## How does fuse-overlayfs compare to vfs in an Eclipse Che CDE?
+
+A series of tests were performed to measure the build time and storage usage differences between the fuse-overlayfs and vfs storage drivers in an Eclipse Che CDE. For this blog post, I will present the measurements for building a series of different GitHub projects. If you're interested, the results for tests that were specifically designed to highlight the characteristics of each storage driver are available in this https://github.com/dkwon17/storage-driver-test/blob/main/results.md[GitHub repository].
+
+All tests presented in Fig. 2. were run on an OpenShift Dedicated 4.15.3 cluster with a productized version of Eclipse Che 7.82.1. 
+
+[cols="1,1"]
+|===
+|Test name |Project URL 
+
+|dashboard
+|https://github.com/eclipse-che/che-dashboard[che-dashboard]
+
+|dashboard-edit
+|https://github.com/eclipse-che/che-dashboard[che-dashboard]
+
+|operator
+|https://github.com/eclipse-che/che-operator[che-operator]
+
+|operator-edit
+|https://github.com/eclipse-che/che-operator[che-operator]
+
+|server
+|https://github.com/eclipse-che/che-server[che-server]
+
+|server-edit
+|https://github.com/eclipse-che/che-server[che-server]
+
+|nginx-alpine
+|https://github.com/nginxinc/docker-nginx/tree/master/stable/alpine[docker-nginx]
+
+|quarkus-api-example
+|https://github.com/che-incubator/quarkus-api-example[quarkus-api-example]
+|===
+
+Fig. 2. Table of test names name project URLs.
+
+For each test apart from the <projectname>-edit tests, the `podman system reset` command was run before running each test in order to clear the graphroot directory. This removes all image layers, therefore these tests measure a clean-build scenario, just like when a user creates and runs `podman build` in a new CDE.
```

Comment:
it's not completely clear what does exactly user creates in phrase `when a user **creates** and runs `podman build` in a new CD`

Reply:
Fixed with:
```
just like when a user creates a CDE and runs `podman build` for the first time.
```

Reply:
I changed it to:
```
With the fuse-overlayfs storage driver, you can enable faster builds and a more optimized storage usage for `podman build` and `buildah` within your Eclipse Che cloud development environment (CDE). Before diving into its advantages, let's first discuss some prerequisite details about container image layers and storage drivers.
```